buildenv/Build_Instructions_V10.md

```diff
@@ -0,0 +1,431 @@
+
+1. Install the list of dependencies that can be obtained with the `apt-get` command from the following section of the Dockerfile:
+```
+apt-get update \
+  && apt-get install -qq -y --no-install-recommends \
+    autoconf \
+    ca-certificates \
+    ...
+```
+
+2. This build uses the same gcc and g++ compiler levels as OpenJDK, which might be
+backlevel compared with the versions you use on your system. Create links for
+the compilers with the following commands:
+```
+ln -s g++ /usr/bin/c++
+ln -s g++-4.8 /usr/bin/g++
+ln -s gcc /usr/bin/cc
+ln -s gcc-4.8 /usr/bin/gcc
+```
+
+3. Download and setup **freemarker.jar** into a directory. The example commands use `/root` to be consistent with the Docker instructions. If you aren't
+using Docker, you probably want to store the **freemarker.jar** in your home directory.
+```
+cd /root
+wget https://sourceforge.net/projects/freemarker/files/freemarker/2.3.8/freemarker-2.3.8.tar.gz/download -O freemarker.tgz
+tar -xzf freemarker.tgz freemarker-2.3.8/lib/freemarker.jar --strip=2
+rm -f freemarker.tgz
+```
+
+4. Download and setup the boot JDK using the latest AdoptOpenJDK v9 build.
+```
+cd /root
+wget -O bootjdk9.tar.gz https://api.adoptopenjdk.net/openjdk9-openj9/releases/x64_linux/latest/binary
+tar -xzf bootjdk9.tar.gz
+rm -f bootjdk9.tar.gz
+ls | grep -i jdk | xargs -I % sh -c 'mv % bootjdk9'
+
+export JAVA_HOME="/root/bootjdk9"
+export PATH="${JAVA_HOME}/bin:${PATH}"
+```
+
+### 2. Get the source
+:penguin:
+First you need to clone the Extensions for OpenJDK for OpenJ9 project. This repository is a git mirror of OpenJDK without the HotSpot JVM, but with an **openj9** branch that contains a few necessary patches. Run the following command:
+```
+git clone https://github.com/ibmruntimes/openj9-openjdk-jdk10.git
+```
+Cloning this repository can take a while because OpenJDK is a large project! When the process is complete, change directory into the cloned repository:
+```
+cd openj9-openjdk-jdk10
+```
+Now fetch additional sources from the Eclipse OpenJ9 project and its clone of Eclipse OMR:
+```
+bash get_source.sh
+```
+
+### 3. Configure
+:penguin:
+When you have all the source files that you need, run the configure script, which detects how to build in the current build environment.
+```
+bash configure --with-freemarker-jar=/root/freemarker.jar
+```
+:warning: You must give an absolute path to freemarker.jar
+
+### 4. Build
+:penguin:
+Now you're ready to build **OpenJDK V10** with OpenJ9:
+```
+make all
+```
+:warning: If you just type `make`, rather than `make all` your build will fail, because the default `make` target is `exploded-image`. If you want to specify `make` instead of `make all`, you must add `--default-make-target=images` when you run the configure script. For more information, read this [issue](https://github.com/ibmruntimes/openj9-openjdk-jdk10/issues/34).
```

Comment:
The open issue is on jdk9, the jdk10 link is invalid.

Reply:
Not sure if this is still an issue for jdk10

Reply:
for JDK10, `bash configure --help` yields
```
  --with-default-make-target
                          set the default make target [exploded-image]
```
So, this issue still persists in JDK10.